editor/src/clj/editor/collection_non_editable.clj

```diff
@@ -211,24 +209,27 @@
     (distinct)
     instance-property-descs))
 
-(defn- collection-desc->referenced-collection-proj-paths [collection-desc]
-  (eduction
-    (map :collection)
-    (distinct)
-    (:collection-instances collection-desc)))
+(defn- collection-desc->referenced-collection-proj-path->index [collection-desc]
+  (into {}
+        (comp (map :collection)
+              (distinct)
+              (map-indexed flipped-pair))
+        (:collection-instances collection-desc)))
```

Comment:
We now need the scenes from the referenced collections, and contrary to their build targets, we need to be able to figure out which reference corresponds to which source scene so that we can instantiate them in our scene with a transform applied.